rmw_zenoh_cpp/src/detail/rmw_data_types.hpp

```diff
@@ -131,4 +129,78 @@ struct rmw_subscription_data_t
   std::condition_variable * condition{nullptr};
 };
 
+
+///==============================================================================
+
+// z_owned_closure_query_t
+void service_data_handler(const z_query_t * query, void * service_data);
+
+void client_data_handler(z_owned_reply_t * reply, void * client_data);
+
+
+struct saved_queryable_data
+{
+  explicit saved_queryable_data(z_owned_query_t query);
+  ~saved_queryable_data();
+
+  z_owned_query_t query;
+};
+
+///==============================================================================
+
+struct rmw_service_data_t
+{
+  unsigned int get_new_uid();
+
+  const char * zn_queryable_key;
+  z_owned_queryable_t zn_queryable;
+
+  const void * request_type_support_impl;
+  const void * response_type_support_impl;
+  const char * typesupport_identifier;
+  RequestTypeSupport * request_type_support;
+  ResponseTypeSupport * response_type_support;
+
+  rmw_context_t * context;
+
+  // Map to store the query id and the query.
+  // The query handler is saved as it is needed to answer the query later on.
+  std::unordered_map<unsigned int, std::unique_ptr<saved_queryable_data>> id_query_map;
```

Comment:
Is there a reason this map cannot simply be
```suggestion
  std::unordered_map<unsigned int, z_owned_queryable_t> id_query_map;
```

Reply:
Originally I used a custom struct `saved_queryable_data` due to I was expecting more data to be added but it wasn't necessary. I removed the struct.